TempleEight/docs/guide/getting-started.md

```diff
@@ -0,0 +1,281 @@
+> If you'd like to verify this, run `temple --version`, output should be `temple X.X.X (c) 2020 TempleEight`
+
+
+## Project Setup
+
+Temple projects start their life as an empty directory. Let's make one for this tutorial:
+
+```
+~/Documents ❯❯❯ mkdir temple-tutorial
+~/Documents ❯❯❯ cd temple-tutorial
+~/Documents/temple-tutorial ❯❯❯
+```
+
+## Templefile 
+
+The Templefile is the heart of the project, it contains all of the information the Temple CLI uses to build your application files.
+
+A Templefile is any text file with the `.temple` extension. 
+
+For a full reference of the Templefile language, check out the [Templefile Spec](templefile-spec.md).
+
+We'll build an example Templefile from the ground up with minimal features, and add more over the rest of the guides.
+
+### Project Block
+
+```
+ExampleProject: project {
+  #language(go);
+  #database(postgres);
+  #provider(dockerCompose);
+}
+```
+
+The first thing required in a Templefile is the project block. This instantiates the whole project to Temple and gives some project-level metadata.
+
+At the top level, we write `ExampleProject: project`, which defines a new `project` called `ExampleProject`.
+This is the name Temple uses for your project globally.
+
+Then, inside the project block there are a number of metadata items that tell us things about the project on a global scale.
+
+Metadata definitions begin with `#` characters and tell us something about a project or a service, and how it interacts with other services. 
+For a full list of all valid metadata, see the [Templefile Spec](templefile-spec).
+
+Here, we define the server language for all services to be Golang, and the database backing everything up to be Postgres.
+While these two blocks are defined on the project level, they can also be overridden at the service level, if you for example wanted a certain service to be in a different language.
+The final metadata item is the `#provider` block, which specifies that all of the services in this project should be orchestrated with `docker-compose`.
+This is a project level item only, and can't be overridden at the service level.
+
+### Service Blocks
+
+```
+ExampleService: service {
+  foo: string;
+  bar: int;
+}
+```
+
+The next thing to add is a Service Block.
+Each service block in a Templefile defines one generated microservice. 
+
+The opening statement `TestService: service` defines a new service, named `ExampleService`, which is the name Temple will use to refer to this service.
+
+Service block can contain two kinds of declarations: *Property definitions* and *metadata*. 
+
+Property declarations tell us about the kind of data belonging to the entity this service describes. 
+For example, here we say that `ExampleService` has a `foo`  which is a `string` property, and a `bar`, which is an `int` property. 
+Property declarations can either be a [*primitive* datatype](primitives), or be a [*foreign key*](cross-service-coms) to another service.
```

Comment:
Do these links work? Aren’t they in different folders?

Reply:
Yup they work, Docusaurus does some magic with resolving the document ids